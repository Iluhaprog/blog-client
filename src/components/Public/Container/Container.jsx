import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ContainerBox = styled.div`
  max-width: ${({maxWidth}) => maxWidth}px;
`;

const Container = ({maxWidth, children}) => {
  return (
    <ContainerBox maxWidth={maxWidth}>
      {children}
    </ContainerBox>
  );
};

Container.defaultProps = {
  maxWidth: 1280,
};

Container.propTypes = {
  maxWidth: PropTypes.number,
  children: PropTypes.node,
};

export {Container};
