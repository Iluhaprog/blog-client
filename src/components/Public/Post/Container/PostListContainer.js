import {getVisible} from '../../../../store/post/postActions';
import {connect} from 'react-redux';
import {PostList} from '../List';
import {addBookmark} from '../../../../store/bookmark/bookmarkActions';

const mapStateToProps = (state) => ({
  total: state.post.total,
  posts: state.post.posts,
  lang: state.settings.lang,
});

const mapDispatchToProps = (dispatch) => ({
  getPosts: (page) => {
    dispatch(getVisible(page));
  },
  addToBookmarks: (data) => {
    dispatch(addBookmark(data));
  },
});

const PostListContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(PostList);

export {PostListContainer};
