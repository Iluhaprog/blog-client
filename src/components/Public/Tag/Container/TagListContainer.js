import {TagList} from '../List';
import {connect} from 'react-redux';
import {getAll} from '../../../../store/tag/tagActions';

const mapStateToProps = (state) => ({
  tags: state.tag.tags,
});

const mapDispatchToProps = (dispatch) => ({
  getTags: () => {
    dispatch(getAll());
  },
});

const TagListContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(TagList);

export {TagListContainer};
