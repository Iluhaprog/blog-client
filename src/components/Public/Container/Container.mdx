---
name: Container
route: /container
---

import {Playground, Props} from 'docz';
import {Container} from './Container';

This component is for wrapping children
into the container with the maximum width.

<Props of={Container} />

## Basic usage
<Playground>
  <Container maxWidth={600}>
    Container
  </Container>
</Playground>
