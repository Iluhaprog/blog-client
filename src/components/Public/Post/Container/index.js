import {PostListContainer} from './PostListContainer';
export {PostListContainer};
