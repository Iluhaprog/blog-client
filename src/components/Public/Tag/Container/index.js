import {TagListContainer} from './TagListContainer';
export {TagListContainer};
