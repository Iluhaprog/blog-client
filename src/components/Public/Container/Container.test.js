import {render} from '@testing-library/react';
import {Container} from './Container';

test('Should render Container component', () => {
    const {asFragment} = render(
      <Container>
        Test child
      </Container>
    );
    expect(asFragment()).toMatchSnapshot();
});
