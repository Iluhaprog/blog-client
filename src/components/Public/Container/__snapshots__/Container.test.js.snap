// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Container component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ drfgAN"
  >
    Test child
  </div>
</DocumentFragment>
`;
